src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php

```diff
@@ -118,4 +223,211 @@ public function iShouldBeNotifiedThatItHasBeenSuccessfullyCreated(): void
+            'Product still exists, but it should not'
+        );
+    }
+
+    /**
+     * @Then I should see :numberOfProducts products in the list
+     */
+    public function iShouldSeeProductsInTheList(int $count): void
+    {
+        Assert::count($this->responseChecker->getCollection($this->client->getLastResponse()), $count);
+    }
+
+    /**
+     * @Then I should( still) see a product with :field :value
+     */
+    public function iShouldSeeProductWith(string $field, string $value): void
+    {
+        Assert::true(
+            $this
+                ->responseChecker
+                ->hasItemWithTranslation($this->client->getLastResponse(), 'en_US', $field, $value)
+            ,
+            sprintf('Product has not %s with %s', $field, $value)
+        );
+    }
+
+    /**
+     * @Then I should not see any product with :field :value
+     */
+    public function iShouldNotSeeAnyProductWith(string $field, string $value): void
+    {
+        $this->client->update();
+
+        Assert::false(
+            $this
+                ->responseChecker
+                ->hasItemWithTranslation($this->client->getLastResponse(), 'en_US', $field, $value)
+            ,
+            sprintf('Product with %s set as %s still exists, but it should not', $field, $value)
+        );
+    }
+
+    /**
+     * @Then I should not be able to edit its code
+     */
+    public function iShouldNotBeAbleToEditItsCode(): void
+    {
+        $this->client->addRequestData('code', '_NEW');
+        $response = $this->client->index();
+
+        Assert::false(
+            $this->responseChecker->hasItemOnPositionWithValue(
+                $response,
+                0,
+                'code',
+                '/new-api/produ-apicts/_NEW'
+            ),
+            sprintf('It was possible to change %s', '_NEW')
+        );
+    }
+
+    /**
+     * @Then /^(this product) name should be "([^"]+)"$/
+     */
+    public function thisProductNameShouldBe(ProductInterface $product, string $name): void
+    {
+        $response = $this->client->index();
+
+        Assert::true(
+            $this->responseChecker->hasItemWithTranslation($response, 'en_US', 'name', $name),
+            sprintf('Product with name %s does not exist', $name)
+        );
+    }
+
+    /**
+     * @Then /^(this product) should not exist in the product catalog$/
+     */
+    public function productShouldNotExist(ProductInterface $product): void
+    {
+        $response = $this->client->index();
+
+        Assert::false(
+            $this->responseChecker->hasItemWithValue($response, 'code', $product->getCode()),
+            sprintf('Product with name %s still exists, but it should not', $product->getName())
+        );
+    }
+
+    /**
+     * @Then /^this product should have (?:a|an) ("[^"]+" option)$/
+     */
+    public function thisProductShouldHaveOption(ProductOptionInterface $productOption): void
+    {
+        $response = $this->client->index();
```

Comment:
Same about index vs show page as above. This step may throw an exception if there will be some additional product created